pkg/cluster/defaults_test.go

```diff
@@ -226,6 +226,29 @@ func TestNewControlPlaneIPCheckAnnotationDefaulterAddAnnotation(t *testing.T) {
 	g.Expect(oldCluster).To(Equal(clusterSpec.Config.Cluster))
 }
 
+func TestNewClusterNamespaceDefaulter(t *testing.T) {
```

Comment:
I would make this a table test and have one case for empty namespace and another one for not empty